src/vertex_array_object.rs

```diff
@@ -17,12 +16,13 @@ use gl;
 use context::CommandContext;
 use version::Api;
 use version::Version;
+use utils::fnvmap::*;
 
 /// Stores and handles vertex attributes.
 pub struct VertexAttributesSystem {
     // we maintain a list of VAOs for each vertexbuffer-indexbuffer-program association
     // the key is a (buffers-list-with-offset, program) ; the buffers list must be sorted
-    vaos: RefCell<HashMap<(Vec<(gl::types::GLuint, usize)>, Handle), VertexArrayObject>>,
+    vaos: RefCell<FnvHashMap<(Vec<(gl::types::GLuint, usize)>, Handle), VertexArrayObject>>,
```

Comment:
Please leave the regular `HashMap` here, as it is likely faster than the fnv hashmap.